fix: track all events

iamacook opened the pull request:
## What it solves
Adding previously GA-tracked events, as well as newly requested ones from [this](https://docs.google.com/document/d/1RY9oaRV4x8N0wsWrU1H0O_EpQOn9pmF2wq25l0H-EKc/edit#) document.

## How this PR fixes it
All events that were previously being tracked, as well as those that were requested are now added to the `dataLayer`, routed onto GA from GTM.

GTM only initializes/tracks when in the production environment or the debug window is open.

## Notes
There are three event types: `pageview`, `customClick` and `metadata`:
- `pageview` emits the URL, with an anonymized Safe address/transaction ID
- `customClick` emits when an element is clicked (or is a direct side-effect of a click)
- `metadata` emits with associated data, e.g. threshold of a newly created Safe

## How to test it
The checklist of events [here](https://github.com/gnosis/safe-react/issues/3408) will log to the console.

katspaugh commented on src/routes/safe/components/Balances/Coins/index.tsx:
```diff
@@ -79,6 +83,17 @@ const Coins = (props: Props): React.ReactElement => {
+  const tokenAmount = useMemo(() => safeTokens.size, [safeTokens])
+  useEffect(() => {
+    if (tokenAmount > 0) {
+      trackEvent({ ...ASSETS_EVENTS.TOKEN_AMOUNT, label: tokenAmount })
```
This is... controversial. There'll be tons of events with very different amounts.

iamacook replied:
I did raise some concerns about this originally. @johannesmoormann, it would be helpful to debrief on this.

katspaugh replied:
After a discussion: in future we can limit it to once per session, but it's fine for now.

iamacook replied:
There's a few events that are over-tracking. I'm going to look into these now.

katspaugh reviewed (approved):
🥇

usame-algan commented on src/utils/__tests__/googleTagManager.test.tsx:
```diff
@@ -3,13 +3,16 @@ import { matchPath } from 'react-router-dom'
   matchPath: jest.fn(),
 }))
 
+customElements.define('debug-badge', HTMLElement)
```
What is this used for?

iamacook replied:
GTM's `<debug-badge>` web component (injected when previewing GTM) is [used as a flag](https://github.com/gnosis/safe-react/blob/26443542d37c915ce34add9c938f49e3b555857a/src/utils/googleTagManager.ts#L64) to initialise GTM.

iamacook commented on src/routes/safe/components/Settings/SpendingLimit/NewLimitModal/Review.tsx:
```diff
@@ -223,6 +225,10 @@ export const ReviewSpendingLimits = ({ onBack, onClose, txToken, values }: Revie
 
       dispatch(createTransaction({ ...spendingLimitTxData, delayExecution }))
+
+      if (values.withResetTime) {
```
`resetTime` is always there when the `withResetTime` radio is checked. It has a default value of 5 minutes.

iamacook replied:
[This event](https://github.com/gnosis/safe-react/blob/26443542d37c915ce34add9c938f49e3b555857a/src/utils/events/settings.ts#L70) is tracked when clicking on remove and then you can follow the [`'modals'` flow](https://github.com/gnosis/safe-react/blob/26443542d37c915ce34add9c938f49e3b555857a/src/components/Modal/index.tsx#L238) of, e.g. 'Next', 'Review', 'Submit' to guage that it was removed.

DiogoSoaress replied:
Tracking for "no reset time" Spending Limit was added 💪

iamacook commented:
Note: in the last commit I removed the `SHOULD_LOAD_GTM` as it was intefering with preview mode. It is no longer needed as we do not push to the `dataLayer` unless it is production, and when in preview a debug flag is sent to GA.

DiogoSoaress reviewed (approved):
LGTM! 🏷️  ✅

francovenica commented:
When switching accoutns int the MM extension 2 metadata entries are being created that are the same. Not sure if it is a bug or not, just weird behavior
![image](https://user-images.githubusercontent.com/4503659/158999240-d47d122f-81d9-4710-9019-1302ccf592ad.png)

francovenica commented:
The rest of the events are fine,  they are being tracked.

Note that I only checked in the GTM that the events were tracked but not in GA yet, and also we should have a meeting to discuss the integrity of the data being tracked, to make sure we are tracking what we expect.

iamacook commented:
@francovenica, thanks for the thorough testing. I have fixed/adjusted your findings accordingly (pleasse see the checkboxes). Regarding the following:

> If you click that button it will create a metadata log with the value you input in the searchbar, which I don't think is correct. I think it should be something lile "WC by suggestion" or something like that.

I have discussed this with @johannesmoormann and the current implementation is as intended. Should we want to adjust this, we will do so in the future.

As a side note: I have added an extra event when clicking the "Download on the App Store" button, as per @sche's request. It would be appreciated if you can double-check that for me as well.

francovenica commented:
The double click event when opening an app was fixed (the event with the "unknown" eventLabel was removed)
The pin/unpin events were fixed
I checked the new event when clicking the "download in app store" button that is in the "Connect dropdown"
The download in app store button has the link it was requested to have:
![image](https://user-images.githubusercontent.com/4503659/159262831-13d4d33e-1ad9-4090-b750-cc57341c9beb.png)

Looks good to me
